packages/donor/src/screens/bookDonation/BookDonationScreenContainer.tsx

```diff
@@ -34,22 +42,59 @@ export function BookDonationScreenContainer({
   }
 
   const onSlotSelected = (donationSlot: DonationSlotToBook) => {
-    appointmentToBookStore.setAppointmentToBook(donationSlot);
-    if (isLoggedIn) {
-      navigate(MainNavigationKeys.Questionnaire);
+    setSelectedSlot(donationSlot);
+    const currentDonor = appState.donor;
+    const timeDifference = currentDonor?.lastCompletedDonationTimeMillis
+      ? DateUtils.getNumberOfDaysBetweenDates(
+          new Date(donationSlot.donationStartTimeMillis),
+          new Date(currentDonor.lastCompletedDonationTimeMillis)
+        )
+      : MIN_DAYS_BETWEEN_DONATIONS + 1;
+
+    if (timeDifference < MIN_DAYS_BETWEEN_DONATIONS) {
+      setShowWarningPopup(true);
     } else {
-      navigate(MainNavigationKeys.Register);
+      continueDonation();
+    }
+  };
+
+  const continueDonation = () => {
+    if (selectedSlot) {
+      appointmentToBookStore.setAppointmentToBook(selectedSlot);
+      if (isLoggedIn) {
+        navigate(MainNavigationKeys.Questionnaire);
+      } else {
+        navigate(MainNavigationKeys.Register);
+      }
     }
   };
 
   return (
-    <BookDonationScreen
-      availableAppointments={availableAppointmentsStore.availableAppointments}
-      onSlotSelected={onSlotSelected}
-      firstName={appState.donor?.firstName}
-      isFetching={availableAppointmentsStore.isFetching}
-      defaultHospital={""}
-    />
+    <div>
+      <BookDonationScreen
+        availableAppointments={availableAppointmentsStore.availableAppointments}
+        onSlotSelected={onSlotSelected}
+        firstName={appState.donor?.firstName}
+        isFetching={availableAppointmentsStore.isFetching}
+        defaultHospital={""}
+      />
+      <Popup
```

Comment:
The popup should be inside the screen, not in the container.
The container should only have logic

Reply:
Will move the popup but not sure I understood to the fullest "The container should only have logic", when should we add containers and what code should be there ?